src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1102,12 +1102,62 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                 // Don't lower if we're not emitting or if there were errors. 
                 // Methods that had binding errors are considered too broken to be lowered reliably.
-                if (_moduleBeingBuiltOpt == null || hasErrors)
+                if (_moduleBeingBuiltOpt == null || hasErrors || isEmptySynthesizedStaticConstructor(processedInitializers.BoundInitializers))
```

Comment:
>isEmptySynthesizedStaticConstructor(processedInitializers.BoundInitializers) [](start = 65, length = 76)

It feels like the check is making us to return too early. I think we shouldn't emit the body, but should still go through lowering and other analysis we might do.